core/src/main/java/uk/gov/justice/services/core/sender/SenderProducer.java

```diff
@@ -23,19 +24,23 @@
 @ApplicationScoped
 public class SenderProducer {
 
-    @Inject
-    JmsSenderFactory jmsSenderFactory;
 
-    @Inject
-    ComponentDestination componentDestination;
+    private JmsSenderFactory jmsSenderFactory;
+    private ComponentDestination componentDestination;
+    private DispatcherProducer dispatcherProducer;
 
-    @Inject
-    DispatcherProducer dispatcherProducer;
+    private final Map<Component, Sender> senderMap = new ConcurrentHashMap<>();
 
-    private Map<Component, Sender> senderMap;
+    //for CDI proxying only
+    protected SenderProducer() {
```

Comment:
I think we agreed a while ago to use package scoped injected fields, not constructor injection, as the lesser of all the evils to avoid adding code specifically for CDI. I think we should stick to that convention - it seems common in the JEE world.